cub/cub/block/specializations/block_histogram_atomic_warp_aggregated.cuh

```diff
@@ -0,0 +1,141 @@
+/******************************************************************************
+ * Copyright (c) 2025-, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the NVIDIA CORPORATION nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
+
+/**
+ * @file
+ * The cub::BlockHistogramAtomicWarpAggregated class provides atomic-based
+ * methods for constructing block-wide histograms from data samples
+ * partitioned across a CUDA thread block.
+ */
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#include <cuda/atomic>
+
+#include <type_traits>
+
+#include <cooperative_groups.h>
+
+#include <cooperative_groups/reduce.h>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+CUB_NAMESPACE_BEGIN
+namespace detail
+{
+
+namespace cg = cooperative_groups;
+
+template <int ITEMS_PER_THREAD, int BINS, bool HISTOGRAM_IN_SMEM = false, bool SATURATE = true>
+struct BlockHistogramAtomicWarpAggregated
+{
+  struct TempStorage
+  {};
+
+  _CCCL_DEVICE _CCCL_FORCEINLINE BlockHistogramAtomicWarpAggregated(TempStorage&) {}
+
+  using scope_type = std::conditional_t<HISTOGRAM_IN_SMEM, ::cuda::thread_scope_block, cuda::thread_scope_device>;
```

Comment:
Important: please consistenly use type traits from libcu++. This ensures our headers can be compiled with NVRTC.
```suggestion
  using scope_type = ::cuda::std::conditional_t<HISTOGRAM_IN_SMEM, ::cuda::thread_scope_block, cuda::thread_scope_device>;
```